src/main/java/sorald/cli/Cli.java

```diff
@@ -0,0 +1,163 @@
+        PrettyPrintingStrategy prettyPrintingStrategy = PrettyPrintingStrategy.SNIPER;
+
+        @CommandLine.Option(
+                names = Constants.ARG_SYMBOL + Constants.ARG_FILE_OUTPUT_STRATEGY,
+                description =
+                        "Mode for outputting files: 'CHANGED_ONLY', which means that only changed files will be created in the workspace, and 'ALL', which means that all files, including the unchanged ones, will be created in the workspace.")
+        FileOutputStrategy fileOutputStrategy = FileOutputStrategy.CHANGED_ONLY;
+
+        @CommandLine.Option(
+                names = Constants.ARG_SYMBOL + Constants.ARG_MAX_FIXES_PER_RULE,
+                description = "Max number of fixes per rule.")
+        int maxFixesPerRule = Integer.MAX_VALUE;
+
+        @CommandLine.Option(
+                names = Constants.ARG_SYMBOL + Constants.ARG_REPAIR_STRATEGY,
+                description =
+                        "Type of repair strategy. DEFAULT - load everything without splitting up the folder in segments, SEGMENT - splitting the folder into smaller segments and repair one segment at a time (need to specify --maxFilesPerSegment if not default)")
+        RepairStrategy repairStrategy = RepairStrategy.DEFAULT;
+
+        @CommandLine.Option(
+                names = Constants.ARG_SYMBOL + Constants.ARG_MAX_FILES_PER_SEGMENT,
+                description =
+                        "Max number of files per loaded segment for segmented repair. It should be >= 3000 files per segment.")
+        int maxFilesPerSegment = 6500;
+
+        @Override
+        public Integer call() throws Exception {
+            validateArgs();
+            SoraldConfig config = createConfig();
+            getRepairProcess(config).repair();
+            return 0;
+        }
+
+        private void validateArgs() {
+            if (maxFilesPerSegment <= 0) {
+                throw new CommandLine.ParameterException(
+                        spec.commandLine(),
+                        Constants.ARG_SYMBOL
+                                + Constants.ARG_MAX_FILES_PER_SEGMENT
+                                + " must be greater than 0");
+            }
+        }
+
+        private SoraldConfig createConfig() {
+            SoraldConfig config = new SoraldConfig();
+            config.addRuleKeys(ruleKeys);
+            config.setOriginalFilesPath(originalFilesPath.getAbsolutePath());
+            config.setWorkspace(soraldWorkspace.getAbsolutePath());
+            if (gitRepoPath != null) {
+                config.setGitRepoPath(gitRepoPath.getAbsolutePath());
+            }
+            config.setPrettyPrintingStrategy(prettyPrintingStrategy);
+            config.setFileOutputStrategy(fileOutputStrategy);
+            config.setMaxFixesPerRule(maxFixesPerRule);
+            config.setMaxFilesPerSegment(maxFilesPerSegment);
+            config.setRepairStrategy(repairStrategy);
+            return config;
+        }
```

Comment:
This is one place where we really lower complexity: picocli does a whole lot for us in parsing. See `Main.initConfig` above for what this used to look like.